src/org/mirah/typer/better_closures.mirah

```diff
@@ -389,27 +434,35 @@ class BetterClosureBuilder
         closures_to_skip.add entry
         next
       end
-      bindingName = "b#{i}"
-      bindingForBlocks.put block, bindingName
+      bindingName = "$b#{i}"
+      bindingForBlocks.put uncloned_block, bindingName
+      parent_scope = MirrorScope(get_scope(block)) 
       adjuster = BindingAdjuster.new(
         self,
         bindingName,
-        MirrorScope(get_scope(block)),
+        parent_scope,
         blockToBindings,
         bindingLocalNamesToTypes)
 
-      adjuster.adjust enclosing_b
-    end
-
-    # ignore closures with no parents, they aren't in the final AST, maybe
-    closures.removeAll closures_to_skip
-
-    i = 0
-    closures.each do |entry: Entry|
-      @@log.fine "insert_closure #{entry.getKey} #{entry.getValue} #{i}"
-      i += 1
-
-      block = Block(entry.getKey)
+      binding_generated = adjuster.adjust enclosing_b, block
```

Comment:
`binding_generated` is never used. Could you revert the change to cause adjust to return a bool, and rm it?